platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java

```diff
@@ -610,13 +603,25 @@ public Response updateFile(@ApiParam(value = "Workspace ID", required = true)
     public void delete(@ApiParam(value = "Workspace ID", required = true)
                        @PathParam("ws-id") String workspace,
                        @ApiParam(value = "Path to a resource to be deleted", required = true)
-                       @PathParam("path") String path)
-            throws NotFoundException, ForbiddenException, ConflictException, ServerException {
+                       @PathParam("path") String path) throws NotFoundException, ForbiddenException, ConflictException, ServerException {
         filesBuffer.addToBufferRecursive(getVirtualFileEntry(workspace, path).getVirtualFile());
 
-        if (!projectManager.delete(workspace, path)) {
-            throw new NotFoundException(String.format("Path '%s' doesn't exist.", path));
-        }
+        projectManager.delete(workspace, path);
+    }
+
+    @DELETE
+    @Path("module/{path:.*}")
```

Comment:
If you are defining parameters with swagger annotations, please don't forget to define `@ApiOperation` and `@ApiReponses`